src/apis/spawnHelpers.jsx

```diff
@@ -112,31 +112,18 @@ export const spawnCreateLink = ({ type, currentPath, originalName, newName, Dial
             .then(Dialogs.close, (err) => { setErrorMessage(err.message) });
 };
 
-// eslint-disable-next-line max-len
-export const spawnEditPermissions = ({ ownerAccess, groupAccess, otherAccess, path, selected, owner, group, Dialogs, setErrorMessage }) => {
-    const command = [
-        "chmod",
-        ownerAccess + groupAccess + otherAccess,
-        path.join("/") + "/" + selected.name
-    ];
-    const permissionChanged = (
-        ownerAccess !== selected.permissions[0] ||
-        groupAccess !== selected.permissions[1] ||
-        otherAccess !== selected.permissions[2]
-    );
-    const ownerChanged = owner !== selected.owner || group !== selected.group;
+export const spawnEditPermissions = ({ mode, path, selected, user, group, Dialogs, setErrorMessage }) => {
+    const permissionChanged = mode !== selected.mode;
+    const ownerChanged = user !== selected.user || group !== selected.group;
 
     Promise.resolve()
             .then(() => {
                 if (permissionChanged)
-                    return cockpit.spawn(command, options);
+                    return cockpit.spawn(["chmod", mode.toString(8), path.join("/") + "/" + selected.name], options);
             })
             .then(() => {
                 if (ownerChanged) {
-                    return cockpit.spawn(
-                        ["chown", owner + ":" + group, path.join("/") + "/" + selected.name],
-                        options
-                    );
+                    return cockpit.spawn(["chown", user + ":" + group, path.join("/") + "/" + selected.name], options);
```

Comment:
This reads a lot better. Now we should also kill the pattern of `path.join("/") + "/" + selected.name` everywhere in the code but that's way too offtopic.